libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -219,25 +219,47 @@ bool ExpressionCompiler::visit(Assignment const& _assignment)
 
 bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 {
-	vector<unique_ptr<LValue>> lvalues;
-	for (auto const& component: _tuple.components())
-		if (component)
+	if (_tuple.isInlineArray())
+	{
+		ArrayType const& arrayType = dynamic_cast<ArrayType const&>(*_tuple.annotation().type);
+		auto components = _tuple.components();
+		
+		m_context << max(u256(32u), arrayType.memorySize());
+		utils().allocateMemory();
+		m_context << eth::Instruction::DUP1;
+	
+		for (unsigned i = 0; i < components.size(); ++i)
 		{
-			component->accept(*this);
-			if (_tuple.annotation().lValueRequested)
+			components[i]->accept(*this);
+			utils().convertType(*components[i]->annotation().type, *arrayType.baseType(), true);
+			components[i]->annotation().type = arrayType.baseType(); //force conversion
```

Comment:
Annotations should never be modified during code generation. Why do you think that this forces the conversion?